source/common/http/codec_helper.h

```diff
@@ -13,9 +13,10 @@ class StreamCallbackHelper {
     }
     ASSERT(high_watermark_callbacks_ > 0);
     --high_watermark_callbacks_;
-    // TODO(alyssawilk) see if we can make this safe for disconnects mid-loop
     for (StreamCallbacks* callbacks : callbacks_) {
-      callbacks->onBelowWriteBufferLowWatermark();
+      if (callbacks) {
+        callbacks->onBelowWriteBufferLowWatermark();
+      }
```

Comment:
This looks fine, because of the prior decision that callback removal sets that slot to nullptr but doesn't remove it from the vector. There's a comment below in this file that says this is done "for performance reasons" but the way this loop works it also needs to be done that way for correctness reasons - otherwise we could skip over the callback just after the deleted one when iterating. Can you update the comment so nobody thinks it is safe to change this behavior?